tests/generation/SettingsTest.py

```diff
@@ -65,3 +69,18 @@ def test_allow_complete_section_mode(self):
             bears_list.sort())
 
         self.assertEqual(['**.html'], files_list)
+
+    def test_generate_ignore_fields(self):
+        project_dir = "/repo"
+        project_files = ['/repo/hello.html', '/repo/main.css',
+                         '/repo/style.css', '/repo/file.html']
+        ignore_globs = ['main.css', 'file.html', '/repo/hello.html']
+
+        extset = get_extensions(project_files)
+
+        lang_files = split_by_language(project_files)
+
+        res = generate_ignore_field(
+            project_dir, lang_files.keys(), extset, ignore_globs)
+
+        self.assertIn('main.css', res)
```

Comment:
Imo, we should add another test case where the `ignore_globs` set is disjoint from `project_files` set and make sure the outcome is an empty list.